pyproject.toml

```diff
@@ -1,7 +1,7 @@
 [project]
 name = "stdatamodels"
 description = "Core support for DataModel classes used in calibration pipelines"
-requires-python = ">=3.6"
+requires-python = ">=3.9"
```

Comment:
JWST currently supports 3.8:
https://github.com/spacetelescope/jwst/blob/master/setup.cfg#L26

Is there anything in stdatamodels that is incompatible with python 3.8? The CI was passing for this version so I'd vote for keeping it.

Reply:
that is fair, but when dependencies drop support for 3.8 then users might have issues with the rest of our dependencies

Reply:
Agreed, we should re-check when we bump dependency versions. At the moment they all appear to support 3.8.

Reply:
The next releases of astropy and numpy will not support 3.8. `numpy` has already dropped 3.8 support in its development branch and `astropy` is working towards doing that as well.

Reply:
However, since `stdatamodels` needs to be released sooner rather than later, the min version of python should be 3.8, not 3.9